### ISSUE TITLE
Unable to connect to a remote DB

### ISSUE THREAD

**Issue by yasirmturk** [medusajs/medusa#4881]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/medusajs/medusa/issues

### Issue Summary

Unable to connect to a remote db, localhost works but none of these work, i have tried
- vercel
- cockroachdb
- neondb

```
Error starting server Error: Failed to establish a connection to PostgreSQL. Please ensure the following is true and try again:
      - You have a PostgreSQL database running
      - You have passed the correct credentials in medusa-config.js
      - You have formatted the database connection string correctly. See below:
      "postgres://[username]:[password]@[host]:[post]/[db_name]" - If there is no password, you can omit it from the connection string
      

      connect ECONNREFUSED 52.57.171.9:5432
    at handlePostgresDatabaseError (/home/zbrdst/nodevenv/shop.zabrdast.com/16/lib/node_modules/@medusajs/utils/dist/common/handle-postgres-database-error.js:17:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

### How can this issue be resolved?

1.
2.
3.
...

### Are you interested in working on this issue?

- [ ] I would like to fix this issue

**Comment by shahednasser** [comment]
Thank you for submitting this issue! Can you specify:

1. What command did you use to create your Medusa backend project? (create-medusa-app or medusa new)
2. How did you set up your connection URL and if possible sharing how it's set in `.env` (feel free to omit sensitive information here like the password)

**Comment by yasirmturk** [comment]
1. I used `create-medusa-app ` and git clone the repo, both are almost same 
2. First it complaint about ssl so I added `?sslmode=require`

Anyway, I just slept over it and today it start working on local; without any changes literally :D but it is still not working on server. This is my .env file

```
NODE_ENV=development
DATABASE_TYPE=postgres
DATABASE_URL=postgres://yasirmturk:<password>@ep-odd-breeze-80644673.eu-central-1.aws.neon.tech/neondb?sslmode=require
```

**Comment by shahednasser** [comment]
Sorry this may be a silly question, but did you make sure that your server is using the correct environment variables (similar to local)? Because I'm not sure what the difference should be for it to not work there 😅

Also can you ensure that you're using the same Medusa versions and everything both locally and your server?

**Comment by yasirmturk** [comment]
That is actually a very good question. 
Cpanel provides its own interface to configure env variables just like heroku, i have provided same values there and also in .env file.. 

and the app seems to be initialized  fine, for example it does not complain about the JWT or Cookie secret.. Just unable to connect to the db server. Another fishy thing is that for 1 hit the whole app runs 3 times as I can see in the error logs attached here

I am also checking with the server guys but there should not be any restriction regarding outgoing connections... Is there any way to access the raw logs? can I enable debug logs or something to dig deep into the problem?
<img width="1226" alt="Screenshot 2023-08-30 at 7 21 54 PM" src="https://github.com/medusajs/medusa/assets/306691/10cff550-8ac1-43b3-8ba9-a40164b993e4">

**Comment by shahednasser** [comment]
The JWT and Cookie secrets aren't an indication since a temporary value can be used if not provided in development, and it seems like you're using the `medusa develop` command, otherwise the logs output would be different. I would start with ensuring that the environment variables are actually loaded.

You can specify a log file by setting the `LOG_FILE` environment variable, but it's currently only supported through environment variables 😅

**Comment by yasirmturk** [comment]
So, I specified `LOG_FILE=log.txt` and this is what I get in the file 
```
{"level":"info","message":"Using fake Redis","timestamp":"2023-08-31 08:35:37"}
{"level":"warn","message":"Local Event Bus installed. This is not recommended for production.","timestamp":"2023-08-31 08:35:37"}
{"level":"info","message":"Using fake Redis","timestamp":"2023-08-31 08:35:40"}
{"level":"warn","message":"Local Event Bus installed. This is not recommended for production.","timestamp":"2023-08-31 08:35:40"}
{"level":"info","message":"Using fake Redis","timestamp":"2023-08-31 08:35:43"}
{"level":"warn","message":"Local Event Bus installed. This is not recommended for production.","timestamp":"2023-08-31 08:35:43"}
```

**Comment by shahednasser** [comment]
That's not really an error. It just means that you should be using a [production redis module](https://docs.medusajs.com/development/events/modules/redis) and ensure you've [setup redis configuraitons](https://docs.medusajs.com/development/backend/configurations#redis_url).

**Comment by yasirmturk** [comment]
So, the port is blocked on the nodejs app for cpanel, the issue has been resolved, thank you for you help but you might want to add it to you documentation.

**Comment by shahednasser** [comment]
Sure, thanks for pointing it out!

**Comment by fffilps** [comment]
@yasirmturk would you mind helping me deploy using cpanel. Im also using it and im struggling to set it up, especially the postgresql portion.

**Comment by devwatts** [comment]
> 1. I used `create-medusa-app ` and git clone the repo, both are almost same
> 2. First it complaint about ssl so I added `?sslmode=require`
> 
> Anyway, I just slept over it and today it start working on local; without any changes literally :D but it is still not working on server. This is my .env file
> 
> ```
> NODE_ENV=development
> DATABASE_TYPE=postgres
> DATABASE_URL=postgres://yasirmturk:<password>@ep-odd-breeze-80644673.eu-central-1.aws.neon.tech/neondb?sslmode=require
> ```

This helped me , Thanks!

**Comment by storefronticu-admin** [comment]
Hi there if you are using the --db url option make sure that the db url is incased in "" or else it will fail

**Comment by portwatcher** [comment]
```
npx create-medusa-app@latest --db-url "postgresql://your_db_connection_url_for_remote_postgre"
```
it works for me

**Comment by njNagaraj** [comment]
C:\Users\Nj\Desktop\Medusa.js>npx create-medusa-app@latest 
? What's the name of your project? medusa-app
? Would you like to create the Next.js storefront? You can also create it later No
? Enter your Postgres username medusajs_db_user
? Enter your Postgres password [hidden]
? Enter your Postgres user's database name medusajs_db
Couldn't connect to PostgreSQL because of the following error: AggregateError.

Make sure you have PostgreSQL installed and the credentials you provided are correct.

If you keep running into this issue despite having PostgreSQL installed, please check out our troubleshooting guidelines: https://docs.medusajs.com/resources/troubleshooting/database-errors
I'm trying to connect my render.com postgres db to my medusa but its not working. why?

**Comment by nyousefzai** [comment]
get the pem file for your RDS 

`  databaseDriverOptions:
      process.env.NODE_ENV === "development"
        ? {
            connection: {
              ssl: {
                ca: fs.readFileSync("./us-east-1-bundle.pem").toString(),
                rejectUnauthorized: true,
              },
            },
          }
        : {},
  },`
